vavr/generator/Generator.scala

```diff
@@ -2011,6 +2010,32 @@ def generateMainClasses(): Unit = {
                 """}
             }
 
+            ${(i < N).gen(xs"""
+              /$javadoc
+               * Append a value to this tuple.
+               *
+               * @param t${i+1} the value to append
+               * @return a new Tuple with the value appended
```

Comment:
@valery1707 @danieldietrich I understand if this is a convention vavr has taken for their javadocs, but this seems odd and confusing to me. It makes it appear that the method takes two parameters when there really is only one parameter. Where does this convention come from? When I look at the javadocs for `java.util.List.add(E e)`, I don't see an additional javadoc line for the `E` type.

Reply:
Because that `<E>` type is documented at the class level since it's the generic type argument for the class itself, the `java.util.List.add(E e)` doesn't introduce a new type argument in the equation. See [`Tuple2.map1`](https://static.javadoc.io/io.vavr/vavr/0.9.2/io/vavr/Tuple2.html#map1-java.util.function.Function-) for an example on what is needed to be documented here.

Reply:
@jlorenzen JDK itself use JavaDoc for generic attibutes.
You can find this at `java.util.stream.Stream#map`: [JavaDoc](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#map-java.util.function.Function-) / [Source](http://hg.openjdk.java.net/jdk8/jdk8/jdk/file/687fd7c7986d/src/share/classes/java/util/stream/Stream.java#l178)